html5ever/src/serialize/mod.rs

```diff
@@ -159,7 +160,10 @@ impl<Wr: Write> Serializer for HtmlSerializer<Wr> {
     }
 
     fn end_elem(&mut self, name: QualName) -> io::Result<()> {
-        let info = self.stack.pop().expect("no ElemInfo");
+        let info = self.stack.pop().unwrap_or_else(|| {
```

Comment:
Rather than unconditionally creating a missing element, I would prefer to add an option to SerializeOpts and only create the parent if this behaviour is specifically requested.